src/test/java/core/basesyntax/db/service/impl/FileParserServiceImplTest.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax.db.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.db.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FileParserServiceImplTest {
+    private static FileParserServiceImpl fileParserService;
+
+    @BeforeClass
+    public static void setUp() {
+        fileParserService = new FileParserServiceImpl();
+    }
+
+    @Test
+    public void parse_validListDataType_ok() {
+        List<String> data = List.of("b,banana,130", "p,banana,15", "r,apple,20");
+        List<FruitTransaction> expected = List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 130),
+                new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 15),
+                new FruitTransaction(FruitTransaction.Operation.RETURN, "apple", 20));
+        List<FruitTransaction> actual = fileParserService.parse(data);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void parse_inputIsEmpty_ok() {
+        List<FruitTransaction> expected = new ArrayList<>();
+        List<FruitTransaction> actual = fileParserService.parse(Collections.emptyList());
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void parse_letterInUpperCase_ok() {
+        List<String> data = new ArrayList<>(List.of("B,apple,20", "b,BANANA,10"));
```

Comment:
```suggestion
        List<String> data = List.of("B,apple,20", "b,BANANA,10"));
```
fix everywhere where you use such a code